Yank/experiment.py

```diff
@@ -1179,58 +1303,156 @@ def _validate_solvents(cls, solvents_description):
+        # Some keys we manually tweak
+        base_solvent_schema = utils.generate_signature_schema(AmberPrmtopFile.createSystem,
+                                                              exclude_keys=['nonbonded_method'])
+        implicit_solvent_default_schema = {'implicit_solvent': base_solvent_schema.pop('implicit_solvent')}
+        rigid_water_default_schema = {'rigid_water': base_solvent_schema.pop('rigid_water')}
+        nonbonded_cutoff_default_schema = {'nonbonded_cutoff': base_solvent_schema.pop('nonbonded_cutoff')}
+        # Cerberus Schema Processing hierarchy:
+        # Input Value -> {default value} -> default setter -> coerce -> allowed/validate
+        #  Handle the use cases for special keys
+        # nonbonded_method
+        base_solvent_schema['nonbonded_method'] = {
+            'allowed': [value for _, value in mapped_openmm_nonbonded_methods.items()],  # Only use valid openmm methods
+            'coerce': to_openmm_app,  # Cast the string first to valid method
+            'required': True,  # This must be set
+            'default': openmm_nonbonded_strings[0],  # Choose a default mapping
+        }
+        # Explicit solvent keys, populate required and dependencies in batch
+        explicit_only_keys = {
+            'clearance': {
+                'type': 'quantity',
+                'coerce': utils.to_unit_validator(unit.angstrom),
+            },
+            'solvent_model': {
+                'type': 'string',
+                'nullable': True,
+                'validator': is_supported_solvent_model,
+                'default_setter': solvent_model_if_explicit_else_none
+            },
+            'positive_ion': {
+                'type': 'string'
+            },
+            'negative_ion': {
+                'type': 'string'
+            },
+            'ionic_strength': {
+                'type': 'quantity',
+                'coerce': to_unit_validator_unless_none(unit.molar),
+                'default_setter': ionic_strength_if_explicit_else_none,
+                'nullable': True
+            },
+            **nonbonded_cutoff_default_schema
+        }
+        # Batch the explicit dependencies
+        for key in explicit_only_keys.keys():
+            explicit_only_keys[key]['dependencies'] = {'nonbonded_method': [mapped_openmm_nonbonded_methods[value] for
+                                                                            value in
+                                                                            get_openmm_explicit_nb_method_strings()]}
+            explicit_only_keys[key]['required'] = False
+
+        # Implicit solvent keys
+        implicit_only_keys = {**implicit_solvent_default_schema}
+        implicit_only_keys['implicit_solvent']['nullable'] = True
+        implicit_only_keys['implicit_solvent']['coerce'] = to_openmm_app_unless_none
+        implicit_only_keys['implicit_solvent']['default_setter'] = implicit_solvent_if_implicit_else_none
```

Comment:
Ah, yes, I missed that when I was reworking the logic. Part copy/paste problem.
I need to change the `default_setter` to detect both nonbonded_method AND the existance of the `implicit_solvent` key in the document, since the default_setter methods have access to the whole document.